### PR TITLE
Auto grow data fs on first boot

### PR DESCRIPTION
**Pull request by om26er**
* Adds data partition with initial size of 500mb
* Optionally Add `rauc-grow-data-part` package, which, if added to `IMAGE_INSTALL_append`, will install a systemd unit that uses `parted` to grow the partition.

Feedback and input welcome.

### CODE REVIEW

File: meta-rauc-raspberrypi/recipes-core/rauc/rauc_%.bbappend [hunk 1]
```diff
@@ -1,5 +1,18 @@
...
-        file://ca.cert.pem \
+	file://ca.cert.pem \
+	file://rauc-grow-data-partition.service \
 "
+
+inherit systemd
+
+SYSTEMD_PACKAGES += " ${PN}-grow-data-part"
+SYSTEMD_SERVICE_${PN}-grow-data-part = "rauc-grow-data-partition.service"
+
```
Comment (dev-0x7C6) [inline meta-rauc-raspberrypi/recipes-core/rauc/rauc_%.bbappend:12]: You may want to add `RDEPENDS_${PN}-grow-data-part += "parted"` to ensure that `parted` will be available at runtime.

File: meta-rauc-raspberrypi/recipes-core/rauc/rauc_%.bbappend [hunk 2]
```diff
@@ -1,5 +1,18 @@
...
+PACKAGES += " rauc-grow-data-part"
```
Comment (dev-0x7C6) [inline meta-rauc-raspberrypi/recipes-core/rauc/rauc_%.bbappend:13]: There is no need empty space when using `+=` operator.

File: meta-rauc-raspberrypi/recipes-core/rauc/rauc_%.bbappend (hunk 1 above)
Comment (om26er) [reply meta-rauc-raspberrypi/recipes-core/rauc/rauc_%.bbappend:12]: Thanks, added that now.

File: meta-rauc-raspberrypi/recipes-core/rauc/rauc_%.bbappend (hunk 2 above)
Comment (om26er) [reply meta-rauc-raspberrypi/recipes-core/rauc/rauc_%.bbappend]: Done ;-)

File: meta-rauc-raspberrypi/wic/sdimage-dual-raspberrypi.wks [hunk 3]
```diff
@@ -1,3 +1,4 @@
 part /boot --source bootimg-partition --ondisk mmcblk0 --fstype=vfat --label boot --active --align 4096 --size 20
 part / --source rootfs --ondisk mmcblk0 --fstype=ext4 --label rootfs_A --align 4096
 part / --source rootfs --ondisk mmcblk0 --fstype=ext4 --label rootfs_B --align 4096
+part /home --source rootfs --rootfs-dir=${IMAGE_ROOTFS}/home --ondisk mmcblk0 --fstype=ext4 --label homefs --align 1024 --size 500 --fsoptions "x-systemd.growfs"
```
Comment (dev-0x7C6) [inline meta-rauc-raspberrypi/wic/sdimage-dual-raspberrypi.wks:4]: I didn't catch that before but when using `${IMAGE_ROOTFS}` inside `wks` file - there is a mandatory requirement to change file name to  `*.wks` to `*.wks.in` in order to expand variables.

1) https://stackoverflow.com/a/56189117
2) https://www.yoctoproject.org/pipermail/yocto/2018-March/040411.html

File: meta-rauc-raspberrypi/wic/sdimage-dual-raspberrypi.wks [hunk 4]
```diff
@@ -1,3 +1,4 @@
 part /boot --source bootimg-partition --ondisk mmcblk0 --fstype=vfat --label boot --active --align 4096 --size 20
 part / --source rootfs --ondisk mmcblk0 --fstype=ext4 --label rootfs_A --align 4096
 part / --source rootfs --ondisk mmcblk0 --fstype=ext4 --label rootfs_B --align 4096
```
Comment (dev-0x7C6) [inline meta-rauc-raspberrypi/wic/sdimage-dual-raspberrypi.wks:3]: Probably we should filter `/home/*` files from being copied to `rootfs_A`/`rootfs_B` using `--exclude-path=home/`. I found that solution [there](https://stackoverflow.com/a/56189117) after reading about `wks.in` on stackoverflow.

1) https://stackoverflow.com/a/56189117

P.S. I need to fix my configuration as well.

Comment (dev-0x7C6) [reply meta-rauc-raspberrypi/wic/sdimage-dual-raspberrypi.wks:3]: After some tests the result is that `--exclude-path=home/` it's excluding home files from `rootfs` as expected (keeping empty `/home` directory for mountpoint). But I was not able to boot to console login somehow (I don't have any system logs - my current setup :-(). So maybe better leave it as is right now, I'll research problem and send patches in future.

File: meta-rauc-raspberrypi/wic/sdimage-dual-raspberrypi.wks (hunk 3 above)
Comment (om26er) [reply meta-rauc-raspberrypi/wic/sdimage-dual-raspberrypi.wks:4]: Is it really needed ? Even without the change this seems to be working fine for our testing.

Comment (om26er) [reply meta-rauc-raspberrypi/wic/sdimage-dual-raspberrypi.wks:4]: Ok, I have now renamed the file to end with `.in`

**Comment by om26er** [comment]
@dev-0x7C6 @leon-anavi I think this is good to land now.

File: meta-rauc-raspberrypi/wic/sdimage-dual-raspberrypi.wks (hunk 3 above)
Comment (dev-0x7C6) [reply meta-rauc-raspberrypi/wic/sdimage-dual-raspberrypi.wks:4]: > Is it really needed ? 

Unfortunately isn't well documented, but from sources that I found and from my experience this is needed. I asked this question in irc (#yocto freenode).

```
[18:21] <dev1990> hi, when I'm using ${IMAGE_ROOTFS} inside image.wks should I rename it to image.wks.in?
[18:22] <dev1990> is it *.in extension to expand variables inside wks ?
[18:24] <dev1990> I can't find any documentation on this topic
[18:26] <kergoth> dev1990: yes, that's correct
```

File: meta-rauc-raspberrypi/wic/sdimage-dual-raspberrypi.wks (hunk 4 above)
Comment (om26er) [reply meta-rauc-raspberrypi/wic/sdimage-dual-raspberrypi.wks:3]: Yeah, I think it makes sense. We should continue as-is and then later visit this add `--exclude-path=home/` parameter

**Comment by om26er** [comment]
Reminder, lets land this please and also add the changes to the dunfell branch as well.

**Comment by leon-anavi** [comment]
Hi,

I don't have time to test it right now but it looks good to me so I will squash and merge it to master for the moment. Thank you again for this contribution.

Thanks,
Leon

File: meta-rauc-raspberrypi/wic/sdimage-dual-raspberrypi.wks (hunk 4 above)
Comment (dev-0x7C6) [reply meta-rauc-raspberrypi/wic/sdimage-dual-raspberrypi.wks:3]: Update: `--exclude-path=home/` is now working correctly for latest gatesgarth branch, https://bugzilla.yoctoproject.org/show_bug.cgi?id=14129

**Pull request merged**